js/src/thankyou.js

```diff
@@ -0,0 +1,354 @@
+        } else {
+            picker_container.hide();
+        }
+    };
+
+    ThankYou.prototype.displayThankedNames = function (show) {
+        var thanked_names_container = this.form.find('.js-thank_you-thanked-names-container');
+        if (show === true) {
+            thanked_names_container.show();
+        } else {
+            thanked_names_container.hide();
+        }
+    };
+
+    ThankYou.prototype.submit = function () {
+        var self = this;
+
+        self.resetErrors();
+
+        var id = self.getThankYouIDInput().val();
+
+        if (id === '') {
+            id = null;
+        }
+
+        var thanked = self.getThankedInput().val();
+        var tags = self.getTagsInput().val();
+        var description = self.getDescriptionInput().val();
+
+        var body = {
+            description: description
+        };
+
+        if (typeof thanked === 'object') {
+            var thanked_array = [];
+            for (var offset in thanked) {
+                var thanked_split = thanked[offset].split('_');
+                thanked_array.push({oclass: parseInt(thanked_split[0]), id: parseInt(thanked_split[1])});
+            }
+            body.thanked = thanked_array;
+        }
+
+        if (typeof tags === 'object') {
+            body.tags = [];
+
+            if (tags !== null) {
+                for (var offset in tags) {
+                    body.tags.push(parseInt(tags[offset]));
+                }
+            }
+        }
+
+        var url = '/api/thankyou/v2/thankyou';
+        if (id !== null) {
+            url += '/' + id;
+        }
+
+
+        $.ajax({
+            url: url,
+            type: 'POST',
+            dataType: 'json',
+            contentType: 'application/json',
+            data: JSON.stringify(body),
+            error: function (response) {
+                var body = response.responseJSON;
+
+                var form_errors = self.form.find('.js-form-error');
+                var problem_details_title_error = form_errors.filter('[data-name="problem_details-title"]');
+                if (problem_details_title_error.length > 0 && 'title' in body) {
+                    self.addError(problem_details_title_error, body.title);
+                }
+
+                if ('invalid-params' in body) {
+                    for (var offset in body['invalid-params']) {
+                        var invalid_param = body['invalid-params'][offset];
+                        if ('name' in invalid_param) {
+                            var error_container = form_errors.filter('[data-name="' + invalid_param.name + '"]');
+                            if (error_container.length > 0 && 'reason' in invalid_param) {
+                                self.addError(error_container, invalid_param.reason);
+                            }
+                        }
+                    }
+                }
+            },
+            success: function (response) {
+                location.reload();
```

Comment:
It's safer to explicitly reference `window.location` rather than relying on the implicit global scope of `window`.

Reply:
Fixed in ef1176accf5c411a39af7103186807ffe773720a

Reply:
Fixed in 9943b657578350c4ff8824119fbe52fe1eb6c88d